templates/sync-to-current.js

```diff
@@ -0,0 +1,96 @@
+// Merge all `package.extend.json` to the template and write to `package.json`.
+
+const { existsSync } = require('fs');
+const fs = require('fs/promises');
+const path = require('path');
+
+const templateJson = require('./package.json');
+
+const dependencyFields = [
+  'dependencies',
+  'devDependencies',
+  'peerDependencies',
+  'optionalDependencies',
+];
+
+const dependencyChanged = (json1, json2) => {
+  return !dependencyFields.every((field) => {
+    if (typeof json1[field] !== typeof json2[field]) {
+      return false;
+    }
+
+    if (typeof json1[field] !== 'object' || json1[field] === null) {
+      return json1[field] === json2[field];
+    }
+
+    const entries1 = Object.entries(json1[field]).sort(([a], [b]) => a.localeCompare(b));
+    const entries2 = Object.entries(json2[field]).sort(([a], [b]) => a.localeCompare(b));
+
+    return (
+      entries1.length === entries2.length &&
+      entries1.every(
+        ([key, value], index) => key === entries2[index][0] && value === entries2[index][1]
+      )
+    );
```

Comment:
Can we simply just sort and concat the entries to a string and compare the string?

Reply:
good